worker-document/src/main/java/com/hpe/caf/worker/document/scripting/JavaScriptEngine.java

```diff
@@ -33,7 +35,16 @@
 
     public JavaScriptEngine()
     {
-        this.scriptEngine = new ScriptEngineManager().getEngineByName("nashorn");
+        this.scriptEngine = GraalJSScriptEngine.create(
+            null,
+            Context.newBuilder("js")
+                .allowExperimentalOptions(true) // Needed for loading from classpath
+                .allowHostAccess(HostAccess.ALL) // Allow JS access to public Java methods/members
+                .allowHostClassLookup(s -> true) // Allow JS access to public Java classes
+                .allowIO(true) // Allow JS IO access to load additional scripts
+                .option("js.load-from-classpath", "true") // Allow JS to load files from the classpath
+                .option("js.syntax-extensions", "true"));
```

Comment:
Are there more options that can be turned on so that changes to the actual scripts are not required?  I sort of half expected to see a "nashorn-compat=true" or something like that.  If there is nothing like that and changes to the scripts must be made then we'll have to treat this as a breaking change - major version number updated, break detailed in release notes, etc.  With the nature of the breaking change it's probably also a knock-on breaking change, i.e. the workers consuming the new framework version are also breaking updates since they no longer support input messages that were supported by the previous version.  We'll have to think carefully about how to roll this out (there is an argument for changing the queue names with a change like this, i.e. allowing the existing versions to consume remaining existing messages).

Reply:
Is this switch useful?
https://github.houston.softwaregrp.net/Verity/darwin-search-and-analytics/blob/v3.3.0/schema/pom.xml#L36

Reply:
Also, like I was saying if possible I'd prefer if we didn't have to change the scripts, but if that's not possible and we have to change them anyway, should we re-evaluate whether we should turn all of these on.  For example I don't know what syntax-extensions=true does but if it is something like using a Nashorn specific for loop construct instead of the JavaScript standard construct then perhaps we should leave it off and just update the scripts to be more standard.  (But like I say that's only a thought for if we have to change the scripts anyway.)

Reply:
There is a flag for backwards compatibility with Nashorn, but I assumed the point of this ticket was to start to move away from all out nashorn specific code (which is a reasonably large amount). But as you point out if that's a breaking change I'll set the compatibility flag on and that should avoid the need to change the scripts.

Reply:
@dermot-hardy  I just tried it out, even with Nashorn compatibility enabled there still needs to be script changes. The way graal.js handles numeric types and function overloading is too different from Nashorn.

Regarding `syntax-extensions` I had been updating the scripts to the correct syntax however I had to switch that on for the workflow as the way we definite and execute the `conditionFunction` of workflow actions is Nashorn specific syntax